functionary/prompt_utils.py

```diff
@@ -0,0 +1,215 @@
+from typing import List, Dict, Any, Optional, Tuple
+import torch
+from enum import Enum
+
+
+class EndToken(str, Enum):
+    system = "<|END_OF_SYSTEM|>"
+    user = "<|END_OF_USER|>"
+    assistant = "<|END_OF_ASSISTANT|>"
+    function = "<|END_OF_FUNCTION_RESULT|>"
+    function_call = "<|END_OF_FUNCTION_CALL|>"
+
+
+def get_end_token_for_message(message: Dict) -> EndToken:
```

Comment:
I believe we can make this @staticmethod of EndToken. 
Something like: from_message

Reply:
will change to static method of EndToken now